lib/decompress/zstd_decompress.c

```diff
@@ -2177,21 +2177,24 @@ size_t ZSTD_decompressStream(ZSTD_DStream* zds, ZSTD_outBuffer* output, ZSTD_inB
                 break;
             }
         case zdss_flush:
-            {   size_t const toFlushSize = zds->outEnd - zds->outStart;
+            if (op != NULL) {
```

Comment:
I like that you blocked off the whole flush operation, rather than just using a ternary on `op += flushedSize`. It's much more clear this way, and I actually didn't consider this approach when I looked at the task :)